### DISCUSSION TITLE
Default Scale is Zero -- Lint for footguns?

### DISCUSSION THREAD

**Discussion by JosephCatrambone** [bevyengine/bevy#5754] [Ideas]
I was having problems with the following code:

```rust
	let mut ssb = SpriteSheetBundle {
		sprite: TextureAtlasSprite::new(0),
		texture_atlas: sprite_sheets.player.clone(),
		transform: Transform { // Eyes here!
			translation: Default::default(),
			rotation: Default::default(),
			scale: Default::default()
		},
		global_transform: Default::default(),
		visibility: Default::default(),
		computed_visibility: Default::default()
	};
```

I thought at first that the player was being drawn past the camera FAR, but this turned out not to be the case.  No matter what I did with the Z-depth, I couldn't make the character appear.  This, however, worked fine.

```rust
	let mut ssb = SpriteSheetBundle {
		sprite: TextureAtlasSprite::new(0),
		texture_atlas: sprite_sheets.player.clone(),
		transform: Default::default(), // Eyes here!
		global_transform: Default::default(),
		visibility: Default::default(),
		computed_visibility: Default::default()
	};
```

Ultimately, it turns out that the problem was nothing to do with the transform and everything to do with the default scale, which is 0,0,0, making the sprite invisible.  

Since scale is a Vec3 I don't think we can change the 'default' for just that type, but perhaps there can be a lint or a debug-only message which helps people who spawn something with a default scale of all zeros?

**Comment by MrGVSV** [comment]
Not related to the linting portion of this, but it's probably recommended to use `..default()` (or `..Default::default()`) for defaulting parts of components/bundles. That way, you get the actual defaults as specified by the struct.

**Reply by JosephCatrambone** [reply]
I was replacing parts of the transform.  The actual bit was 

```
let mut ssb = SpriteSheetBundle {
		sprite: TextureAtlasSprite::new(0),
		texture_atlas: sprite_sheets.player.clone(),
		transform: Transform { // Eyes here!
			translation: Vec3::new(x, y, PLAYER_RENDER_PRIORITY),
			rotation: Default::default(),
			scale: Default::default()
		},
		global_transform: Default::default(),
		visibility: Default::default(),
		computed_visibility: Default::default()
	};
```

**Reply by tim-blackbird** [reply]
You could write that like so:
```rust
Transform {
    translation: Vec3::new(x, y, PLAYER_RENDER_PRIORITY),
    ..Default::default(),
},
```
Or, using `from_xyz`.
```rust
Transform::from_xyz(x, y, PLAYER_RENDER_PRIORITY),
```

**Reply by NiklasEi** [reply]
The same pattern works for the whole bundle. As far as I see, you only need to set two fields.

**Comment by afonsolage** [comment]
There is a unofficial crate for bevy linting: https://github.com/MinerSebas/bevy_lint, but linting at this point of Bevy is very tough work, since API will be changed a lot.

When we got the Bevy Editor I think this kind o problem will be easier to track down, since one could just inspect why the mesh/sprite/UI isn't being rendered.
